modules/HapticGlove_module/src/GloveWearable.cpp

```diff
@@ -329,61 +330,40 @@ bool GloveWearableImpl::getFingertipPoseValues(Eigen::MatrixXd& values)
 
 bool GloveWearableImpl::setFingertipForceFeedbackValues(const std::vector<int>& values)
 {
-    if (values.size() != m_numForceFeedback)
-    {
-        yError() << m_logPrefix
-                 << "size of the force feedback vector is not equal to the size of the default "
-                    "force feedback size.";
-    }
-    for (size_t i = 0; i < values.size(); i++)
-    {
-        std::string fingerName = m_humanFingerNameList[i];
-
-        wearable::msg::WearableActuatorCommand& wearableActuatorCommand
-            = m_iWearActuatorPort.prepare();
-
-        wearableActuatorCommand.value = values[i];
-        wearableActuatorCommand.info.name = m_wearablePrefix
-                                            + wearable::actuator::IHaptic::getPrefix() + fingerName
-                                            + "::ForceFeedback";
-        wearableActuatorCommand.info.type = wearable::msg::ActuatorType::HAPTIC;
-        wearableActuatorCommand.duration = 0;
-
-        m_iWearActuatorPort.write(true); // writeStrict option for wearable haptic device should be
-                                         // set to true to avoid the data loss for all actuators
-    }
     return true;
 }
 
 bool GloveWearableImpl::setFingertipVibrotactileValues(const std::vector<int>& values)
 {
-    if (values.size() != m_numVibrotactileFeedback)
+    return true;
+}
+
+bool GloveWearableImpl::setFingertipHapticFeedbackValues(const std::vector<int>& values)
+{
+    if (values.size() != m_numHapticFeedback)
     {
-        yError()
-            << m_logPrefix
-            << "size of the vibrotactile feedback vector is not equal to the size of the default "
-               "vibrotactile feedback size.";
+        yError() << m_logPrefix
+                 << "size of the haptic feedback vector is not equal to the size of the default "
+                    "haptic feedback size.";
     }
+    wearable::msg::WearableActuatorCommand& wearableActuatorCommand
+        = m_iWearActuatorPort.prepare();
+    wearableActuatorCommand.forceValue.resize(m_numForceFeedback, 0);
+    wearableActuatorCommand.vibroTactileValue.resize(m_numVibrotactileFeedback, 0);
```

Comment:
```suggestion
    wearableActuatorCommand.forceValue.resize(m_numForceFeedback);
    wearableActuatorCommand.vibroTactileValue.resize(m_numVibrotactileFeedback);
```
The values will be set down below anyhow